gallery/tutorials/tutorials/weighted_volume_estimation.py

```diff
@@ -0,0 +1,102 @@
+
+from aspire import downloader
+
+sim_data = downloader.simulated_channelspin()
+
+# This data contains a `Volume` stack, an `Image` stack, weights and
+# corresponding parameters that were used to derive images
+# from the volumes.  For example. the rotations below are the known
+# true simulation projections. In practice these would be derived from
+# an orientation estimation component.
+
+imgs = sim_data["images"]  # Simulated image stack (`Image` object)
+rots = sim_data["rots"]  # True projection rotations (`Rotation` object)
+weights = sim_data["weights"]  # Volume weights (`Numpy` array)
+vols = sim_data["vols"]  # True reference volumes (`Volume` object)
+
+# %%
+# Create a ``ImageSource``
+# ------------------------
+# The image stack and projection rotation (Euler) angles can be
+# associated together during instantiation of an ``ImageSource``.
+# Because this example starts with a dense array of images,
+# an ``ArrayImageSource`` is used.
+
+from aspire.source import ArrayImageSource
+
+src = ArrayImageSource(imgs, angles=rots.angles)
+
+# The images are downsampled for the sake of a quicker tutorial.
+# This line can be commented out to achieve the reference size (54 pixels).
+src = src.downsample(24)
+
+# %%
+# .. note::
+#     This tutorial demonstrates bringing reference data.
+#     It is also possible to just create a ``Simulation`` or use other
+#     ``ImageSource`` objects, so long as the rotations required
+#     for backprojecting are assigned.
+
+# %%
+# Volume Reconstruction
+# ---------------------
+# Performing a weighted volume reconstruction requires defining an
+# appropriate 3D basis and supplying an associated image to volume
+# weight mapping as an array.
+
+from aspire.basis import FFBBasis3D
+from aspire.reconstruction import WeightedVolumesEstimator
+
+# Create a reasonable Basis
+basis = FFBBasis3D(src.L, dtype=src.dtype)
+
+# Setup an estimator to perform the back projections and volume estimation.
+# In this case, the `weights` array comes from the reference data set,
+# and is shaped to map images to volumes.
+print("`weights shape:`", weights.shape)
+estimator = WeightedVolumesEstimator(weights, src, basis, preconditioner="none")
+
+# Perform the estimation, returning a `Volume` stack.
+estimated_volume = estimator.estimate()
+
+# %%
+# .. note::
+#     The ``estimate()`` method requires a fair amount of compute time,
+#     but there should be regularly logged progress towards convergence.
+
+# %%
+# Comparison of Estimated Volume with Source Volume
+# -------------------------------------------------
+# Generate several random projections rotations, then compare these
+# projections between the estimated volumes and the known volumes.
+# If ``src`` was downsampled above, the resulting estimated volumes
+# and projections will be of similar downsampled quality.
+
+from aspire.utils import Rotation, uniform_random_angles
+
+v = 0  # Volume under comparison
```

Comment:
Might be nice to see this comparison for more than one volume, especially since the point is that we successfully reconstructed multiple volumes. OTOH, not sure if the additional projections will look much different for this dataset (not sure how much variation there is between volumes).

Reply:
Just realizing this doesn't make sense since we're not reconstructing volume 1-for-1. Disregard.

Reply:
I've intentionally made it easy for a user can change that line.  The other volumes have different (high frequency) detail/modes.  This tutorial isn't meant to explain the paper, just demonstrate the code usage (which I think it does).